### PR TITLE
Add `list-policy` command

### PR DESCRIPTION
**Pull request by s1ntaxe770r**
this PR implements #103

### CODE REVIEW

**Review by daemon1024 — COMMENTED** [review]
I think we should implement listing/getting policies from KubeArmor in systemd/VM mode before we can merge this.

We already have an plain implementation of getting policies from K8s at https://github.com/kubearmor/kubearmor-client/blob/main/get/get.go ( This was the first command, how I started the project :P )

**Comment by s1ntaxe770r** [comment]
oh yeah I'm yet to finish that. Working on it

**Comment by daemon1024** [comment]
Ooof, Do you think it's okay to put in Draft mode then? That way when you convert it back we know it's ready?

**Comment by Ankurk99** [comment]
@s1ntaxe770r Have you tested this in systemd mode? I am not able to run it. It's showing error: 
image

**Comment by s1ntaxe770r** [comment]
Just realized I had `kubectl proxy` running when I tested this. I need to fix that

**Comment by Ankurk99** [comment]
I am still getting an empty policy table.

**Comment by s1ntaxe770r** [comment]
odd works for me 
image

**Comment by s1ntaxe770r** [comment]
well doesn't work on my ubuntu box now. Trying to fix

**Comment by s1ntaxe770r** [comment]
I figured it out. The formatting of the default yaml present in `/opt/kubearmor/policies` isn't correct so the umarshal doesn't fail but the fields simply do not match, I replaced the file with [this](https://github.com/kubearmor/KubeArmor/blob/main/examples/host-security-policies/hsp-kubearmor-dev-file-dir-allow-fromSource.yaml) and it works 

image

**Comment by Ankurk99** [comment]
> The formatting of the default yaml present in /opt/kubearmor/policies isn't correct 
what do you mean by default yaml in the folder?

**Review by Ankurk99 — CHANGES_REQUESTED** [review]
It's still not working for me. Can you please fix this and I have a few suggestions:
1) Can you also add the pod/container/host name and namespace to which the policies are applied? 
2) PTAL at the go-sec warning
Please re-request a review once you find these to be addressed.

**Comment by Ankurk99** [comment]
I have applied these policies: https://raw.githubusercontent.com/kubearmor/KubeArmor/main/examples/host-security-policies/hsp-kubearmor-dev-proc-path-block.yaml, https://raw.githubusercontent.com/kubearmor/KubeArmor/main/examples/host-security-policies/hsp-kubearmor-dev-file-dir-allow-fromSource.yaml

**Comment by s1ntaxe770r** [comment]
@Ankurk99 this should work now

**Comment by Ankurk99** [comment]
@s1ntaxe770r Yes it's working. Can you please also address the points mentioned [here](https://github.com/kubearmor/kubearmor-client/pull/150#pullrequestreview-1090031837)?

**Comment by s1ntaxe770r** [comment]
@Ankurk99 pod names and namespaces are included in the output , working on the gosec error 
image

**Comment by Ankurk99** [comment]
I still couldn't see this :confused: 
image

**Comment by s1ntaxe770r** [comment]
@Ankurk99  have you pulled the latest changes ?

**Comment by Ankurk99** [comment]
Yes I did.

**Comment by s1ntaxe770r** [comment]
oh wait ,  did you pass the `--namespace` flag? by default `list-policy` would look for policies in `kube-system`. 🤔  should I make namespace a required flag ? @Ankurk99

**Comment by Ankurk99** [comment]
I think, we should be considering all the namespaces for displaying policies. `karmor list-policy` should show all the applied policies in all namespaces.

File: list/list.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+		}
+		return []HostPolicyBackup{}, err
+	}
+	if len(files) <= 0 {
+		fmt.Printf("Unable find host polices")
+		return nil, err
+	}
+	policies := []HostPolicyBackup{}
+	for _, file := range files {
+		path := filepath.Join("/opt/kubearmor/policies", filepath.Clean(file.Name()))
```
Comment (s1ntaxe770r) [inline list/list.go:169]: @Ankurk99  the go sec error seems to be caused by this line , however I am already cleaning the file paths

**Comment by s1ntaxe770r** [comment]
@Ankurk99  this should work for all namespaces now

File: list/list.go (hunk 1 above)
Comment (Ankurk99) [reply list/list.go:169]: I think it's because we are trying to access the path which otherwise would not be possible by a non-root user.

**Comment by Ankurk99** [comment]
> @Ankurk99 this should work for all namespaces now
Still can't find this working. :/
image

Hey @daemon1024 can you also pls try this out?

File: list/list.go (hunk 1 above)
Comment (s1ntaxe770r) [reply list/list.go:169]: oh ,  I guess this is ok? wdyt?

**Comment by yasin-cs-ko-ak** [comment]
As of now it is not necessary to have this command. Because we already have `karmor probe` command and it'll show all applied policies and information. We can close this PR.
Any thoughts?
cc: @Ankurk99 @daemon1024 @nyrahul

**Comment by Ankurk99** [comment]
Closing as k8s based policies already handled in `karmor probe`. We can update the `karmor probe` displaying host based policies.

**Pull request closed without merging**